api/src/Filter/ArchivedFilter.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Filter;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\AbstractFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use App\Entity\ArchivableInterface;
+use Doctrine\ORM\QueryBuilder;
+
+final class ArchivedFilter extends AbstractFilter implements GenericFilterInterface
+{
+    private const PARAMETER_NAME = 'archived';
+
+    /**
+     * Filter entities that have a not null "archivedAt" field value.
+     *
+     * @param array<string, mixed> $context
+     */
+    public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, string $operationName = null, array $context = []): void
+    {
+        $archived = $context['filters'][self::PARAMETER_NAME] ?? null;
+        if ('true' !== $archived && $this->supports($resourceClass)) {
```

Comment:
```suggestion
        if (!is_a($resourceClass, ArchivableInterface::class, true)) {
             throw new \InvalidArgumentException("Can't apply the Archived filter on a resource ({$resourceClass}) not implementing the ArchivableInterface.");
        }

        if (!$this->normalizeValue($context['filters'][self::PARAMETER_NAME] ?? null)) {
            return;
        }
```

Then, no need for `supports` method, nor GenericFilterInterface

Reply:
Yes, indeed much shorter with `is_a`!